src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const formattedDateParts = {
+    YYYY: '',
+    MM: '',
+    DD: '',
+  };
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        formattedDateParts['YYYY'] = dateParts[i].substring(0, 4);
+
+        if (toFormat.includes('YY')) {
+          formattedDateParts['YY'] = dateParts[i].substring(2, 4);
+        }
+        break;
+
+      case 'YY':
+        let year = dateParts[i];
+
+        if (parseInt(year) < 30) {
+          year = '20' + year;
+        } else {
+          year = '19' + year;
+        }
```

Comment:
The logic for converting 'YY' to 'YYYY' is correct, but it's better to use 'parseInt' with a radix of 10 to avoid issues with octal numbers when the string starts with '0'.